great_expectations/checkpoint/checkpoint.py

```diff
@@ -122,21 +122,27 @@ def get_substituted_config(
         if isinstance(config, dict):
             config = CheckpointConfig(**config)
 
+        substituted_config: Union[CheckpointConfig, dict]
         if (
             self._substituted_config is not None
             and not runtime_kwargs.get("template_name")
             and not config.template_name
         ):
-            substituted_config = deepcopy(self._substituted_config)
+            substituted_config = deepcopy(self._substituted_config.to_json_dict())
             if any(runtime_kwargs.values()):
-                substituted_config.update(runtime_kwargs=runtime_kwargs)
+                substituted_config.update(runtime_kwargs)
```

Comment:
Whenever an `update` is needed, we should be calling `.update(runtime_kwargs=runtime_kwargs` on a `CheckpointConfig` object, rather than the vanilla `dict` update, since there are particular update rules that are codified in `CheckpointConfig`

Reply:
This did not work fully, because somehow it appended the same docs site twice -- the update method was not disambiguating; I will look again.

Reply:
Fixed; there were other issues found (as per our discussion, and even more!). :)